File: products-app/src/main/java/com/example/products/service/ProductService.java
package com.example.products.service;

import org.springframework.stereotype.Service;

import com.example.products.dto.ProductDto;

import reactor.core.publisher.Mono;

@Service
public class ProductService {

	public Mono<ProductDto> getProduct(String id) {
		ProductDto dummyProduct = new ProductDto(id, "Test Product " + id);
		return Mono.just(dummyProduct);
	}
}

File: products-app/src/main/kotlin/com/example/products/api/ProductService.kt
package com.example.products.api

import com.example.products.dto.ProductDto
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RestController
import reactor.core.publisher.Mono

@RestController
class ProductService {

    @GetMapping("products/get/{id}")
    fun getProduct(@PathVariable id: String): Mono<ProductDto> {
        val dummyProduct = ProductDto(
                id = id,
                name = "Test Product")
        return Mono.just(dummyProduct)
    }
}